docs/resources/aviatrix_gateway.md

```diff
@@ -288,8 +288,8 @@ The following arguments are supported:
 
 ~> As of Provider version R2.21.2+, the `vpc_id` of an OCI VCN has been changed from its name to its OCID.
 
-~> As of Provider version R.22.0+, the `vpc_id` of a GCP VPC has been updated to include the project name.
-* `vpc_id` - (Required) VPC ID/VNet name of cloud provider. Example: AWS/AWSGov/AWSChina: "vpc-abcd1234", GCP: "vpc-gcp-test~-~project-name", Azure/AzureGov/AzureChina: "vnet_name:rg_name:resource_guid", OCI: "ocid1.vcn.oc1.iad.aaaaaaaaba3pv6wkcr4jqae5f44n2b2m2yt2j6rx32uzr4h25vqstifsfdsq".
+!> As of Provider version R.22.0+, the `vpc_id` of a GCP VPC has been updated to include the project ID, e.g. vpc_name~-~project_id. When creating a Gateway using the old format, referencing `vpc_id` in another resource on the same apply that creates this Gateway will cause Terraform to throw an error. Please use the Gateway data source to reference the `vpc_id` of this Transit Gateway in other resources.
```

Comment:
was it the project ID or name? 

And is there a use-case for using the 'old' format (vpc_name) rather than just referencing `.vpc_id` from the vpc resource?

If the gateway's created before this update referenced `vpc_id` from the VPC resource, there should be no changes needed, right?

Reply:
I checked the GCP console and it is the project ID not the name, so I fixed it here.

For most people, there will be no diff if they just reference the `vpc_id` attribute from the vpc resource. There will be no error when referencing `vpc_id` of the gateway in other resources in this case. Any existing gateways will also not be effected (even if they have the old format hard coded in the Terraform config).

The only time there is an issue is if a customer hard codes the old format(just vpc_name) as the `vpc_id` and tries to create an entire topology in a single apply.